server/src/lib/server.rs

```diff
@@ -25,6 +40,35 @@ use crate::serial::Serial;
 // TODO(error) Do a pass of HTTP codes (error and ok)
 // TODO(idempotency) Idempotency mechanisms?
 
+/// Errors which may occur during the course of a serial connection
+#[derive(Error, Debug)]
+enum SerialTaskError {
+    #[error("Cannot upgrade HTTP request to WebSockets: {0}")]
+    Upgrade(#[from] hyper::Error),
+
+    #[error("WebSocket Error: {0}")]
+    WebSocket(#[from] tungstenite::Error),
+
+    #[error("IO error: {0}")]
+    Io(#[from] std::io::Error),
+}
+
+struct SerialTask {
```

Comment:
(Nitpick)

WDYT about adding a `drop` impl that explicitly calls [`abort`](https://docs.rs/tokio/1.6.1/tokio/task/struct.JoinHandle.html#method.abort)?

Might just be a "me" thing, but I always get bothered that the default behavior of `JoinHandle` is to detach the associated task, rather than abort it, when dropped. However, if (for whatever reason) a `SerialTask` is dropped - maybe we shutdown the entire server while serving a websocket - it seems odd that the serial task would continue execution.

Reply:
Not really needed I think. Once SerialTask gets dropped so too does the oneshot Sender. That will still trip the detach branch in the select loop and it'll try to gracefully close the WebSocket and exit the task.